sawmill-core/src/main/java/io/logz/sawmill/ConditionFactoryRegistry.java

```diff
@@ -6,16 +6,11 @@
 import java.util.Map;
 
 public class ConditionFactoryRegistry {
-    private final Map<String, Condition.Factory> conditionFactory = new HashMap<>();
```

Comment:
Why are we turning this static factory into a non-static one? More to the point, why in this PR?

Reply:
because thats a needed change in order to pass the mmdb location

Reply:
I get that you re-factored this area. But all the changes in one PR make this hard to read. You have to split the PR into smaller chunks that are easier to review.